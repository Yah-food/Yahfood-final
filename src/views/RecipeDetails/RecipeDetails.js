import React, { useEffect, useState } from 'react';
import { getRecipeByTitle, getUserRecipeByTitle } from '../../services/fetchData';
import { Link, useHistory } from 'react-router-dom';
import { deleteRecipe } from '../../services/users';

export default function RecipeDetails(props) {
  const title = props.match.params.title;
  const [recipe, setRecipe] = useState({});
  const [loading, setLoading] = useState(true);
  const history = useHistory();

  useEffect(() => {
    const fetchData1 = async () => {
      const userData = await getUserRecipeByTitle(title);
      setRecipe(userData);
    };
    const fetchData2 = async () => {
      const data = await getRecipeByTitle(title);
      setRecipe(data);
    };
    Promise.all([fetchData1(), fetchData2()]).then((result1, result2) => {
      setLoading(false);
    });
  }, [title, loading]);

  const handleDelete = async (e) => {
    e.preventDefault();
    await deleteRecipe(recipe);
    history.push('/profile');
  };

  return (
    <>
      <div className="pt-2">
        <Link to="/profile" className="mx-5">
          <button>Back</button>
        </Link>
      </div>
      <div className="flex justify-center font-sans text-4xl opacity-70 p-5">
        <h1>{recipe.title}</h1>
      </div>
      <div className="mx-5 shadow-lg opacity-80">
        <img src={recipe.images} />
      </div>
      <div className="flex justify-center my-5 opacity-70">
        <p>{recipe.description}</p>
      </div>
      <div className="flex justify-center mx-5 opacity-70">
        <p>{recipe.instructions}</p>
      </div>
      <Link to={`/profile/editrecipe/${recipe.title}`}>
        <div className="flex justify-center mx-auto my-5 w-64 opacity-70 items-center border-2 border-lime-700/50 rounded-lg bg-lime-700/50">
          <button>Edit</button>
        </div>
      </Link>
      <div className="flex justify-center mx-auto my-5 w-64 opacity-70 items-center border-2 border-red-700/50 rounded-lg bg-red-700/50">
        <button onClick={handleDelete}>Delete</button>
      </div>
    </>
  );
}
