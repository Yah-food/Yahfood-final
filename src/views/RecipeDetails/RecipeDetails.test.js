import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import RecipeDetails from './RecipeDetails';

test.skip('renders the recipe details', async () => {
  // const currentResults = [
  //   {
  //     id: 5,
  //     title: 'Fried Chicken',
  //     images:
  //       'https://www.tasteofhome.com/wp-content/uploads/2018/01/Best-Ever-Fried-Chicken_EXPS_MCMZ16_23240_B07_12_3b.jpg?resize=700,700',
  //     description: `1-3/4 cups all-purpose flour
  //     1 tablespoon dried thyme
  //     1 tablespoon paprika
  //     2 teaspoons salt
  //     2 teaspoons garlic powder
  //     1 teaspoon pepper
  //     1 large egg
  //     1/3 cup 2% milk
  //     2 tablespoons lemon juice
  //     1 broiler/fryer chicken (3 to 4 pounds), cut up
  //     Oil for deep-fat frying`,
  //     instructions: `1. In a shallow bowl, mix the first 6 ingredients. In a separate shallow bowl, whisk egg, milk and lemon juice until blended. Dip chicken in flour mixture to coat all sides; shake off excess. Dip in egg mixture, then again in flour mixture.
  //     2. In an electric skillet or deep fryer, heat oil to 375Â°. Fry chicken, a few pieces at a time, 6-10 minutes on each side or until skin is golden brown and juices run clear. Drain on paper towels.`,
  //   },
  // ];
  const { container } = render(
    <MemoryRouter>
      <RecipeDetails match={{ params: { title: 'Fried Chicken' } }} />
    </MemoryRouter>
  );
  await screen.findByText('Fried Chicken');
  expect(container).toMatchSnapshot();
});
