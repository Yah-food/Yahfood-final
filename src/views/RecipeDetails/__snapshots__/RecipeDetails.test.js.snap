// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the recipe details 1`] = `
<div>
  <div>
    <h1>
      Fried Chicken
    </h1>
    <img
      src="https://www.tasteofhome.com/wp-content/uploads/2018/01/Best-Ever-Fried-Chicken_EXPS_MCMZ16_23240_B07_12_3b.jpg?resize=700,700"
    />
  </div>
  <div>
    <p>
      1-3/4 cups all-purpose flour
1 tablespoon dried thyme
1 tablespoon paprika
2 teaspoons salt
2 teaspoons garlic powder
1 teaspoon pepper
1 large egg
1/3 cup 2% milk
2 tablespoons lemon juice
1 broiler/fryer chicken (3 to 4 pounds), cut up
Oil for deep-fat frying
    </p>
  </div>
  <div>
    <p>
      1. In a shallow bowl, mix the first 6 ingredients. In a separate shallow bowl, whisk egg, milk and lemon juice until blended. Dip chicken in flour mixture to coat all sides; shake off excess. Dip in egg mixture, then again in flour mixture.
2. In an electric skillet or deep fryer, heat oil to 375Â°. Fry chicken, a few pieces at a time, 6-10 minutes on each side or until skin is golden brown and juices run clear. Drain on paper towels.
    </p>
  </div>
  <a
    href="/profile/editrecipe/Fried Chicken"
  >
    <button>
      Edit
    </button>
  </a>
  <button>
    Delete
  </button>
</div>
`;
